ha/setup/create_pacemaker_resources.py

```diff
@@ -19,262 +19,299 @@
 from ha.execute import SimpleCommand
 from cortx.utils.log import Log
 
-
-class CreateResourceError(Exception):
-    """Exception to indicate any failure happened during resource creation."""
-
-
-class CreateResourceConfigError(CreateResourceError):
-    """Exception to indicate that given resource configuration is incorrect or incomplete."""
-
+from ha.core.error import CreateResourceError
+from ha.core.error import CreateResourceConfigError
+from ha import const
+
+"""
+# List of resource
+motr-free-space-mon
+io_group-clone
+    hax-consul
+    hax
+    motr-confd-1
+    motr-ios-1
+    haproxy
+    s3auth
+    s3server-1
+    s3server-2
+    s3server-3
+    s3server-4
+    s3backcons
+s3backprod
+sspl-ll-clone
+    sspl-ll
+management
+    kibana
+    csm-agent
+    csm-web
+"""
+
+process = SimpleCommand()
 
 def cib_push(cib_xml):
     """Shortcut to avoid boilerplate pushing CIB file."""
-    cmd_push = f"pcs cluster cib-push {cib_xml} --config"
-    SimpleCommand().run_cmd(cmd_push)
-
+    process.run_cmd(f"pcs cluster verify -V {cib_xml}")
+    process.run_cmd(f"pcs cluster cib-push {cib_xml} --config")
 
 def cib_get(cib_xml):
     """Generate CIB file using pcs."""
-    cmd = f"pcs cluster cib {cib_xml}"
-    SimpleCommand().run_cmd(cmd)
-
+    process.run_cmd(f"pcs cluster cib {cib_xml}")
     return cib_xml
 
-
-def motr_hax(cib_xml, push=False):
-    """Create resources that belong to motr group and clone the group."""
+def hax(cib_xml, push=False, **kwargs):
+    """Create resources that belong to hax and clone the group."""
     # TODO cmd_hare_consul is temporary code and should be removed once fixed by hax/provisioner.
-    cmd_hare_consul = f"pcs -f {cib_xml} resource create hax-consul systemd:hare-consul-agent --group io_group"
-    cmd_hax = f"pcs -f {cib_xml} resource create hax systemd:hare-hax --group io_group"
-    cmd_confd = f"pcs -f {cib_xml} resource create motr-confd-1 ocf:seagate:dynamic_fid_service_ra service=m0d fid_service_name=confd --group io_group"
-    cmd_ios = f"pcs -f {cib_xml} resource create motr-ios-1 ocf:seagate:dynamic_fid_service_ra service=m0d fid_service_name=ios --group io_group"
-
-    for s in (cmd_hare_consul, cmd_hax, cmd_confd, cmd_ios):
-        SimpleCommand().run_cmd(s)
-
+    # hax-consul
+    process.run_cmd(f"pcs -f {cib_xml} resource create hax-consul systemd:hare-consul-agent \
+        op start timeout=100s interval=0s \
```

Comment:
Looks like some of the timeouts were not configured earlier, now I see those configured. This comment is applicable to other resources below too.
How did we arrive at the timeout values?

Reply:
previously it has picked default time. but from past experience we need more time for timeout so added for each resource.

Reply:
ok